packages/insomnia-app/app/ui/components/error-boundary.js

```diff
@@ -57,10 +73,14 @@ class SingleErrorBoundary extends React.PureComponent<Props, State> {
 
   render() {
     const { error, info } = this.state;
-    const { errorClassName, children } = this.props;
+    const { errorClassName, children, renderError } = this.props;
 
     if (error && info) {
-      return <div className={errorClassName || null}>Render Failure: {error.message}</div>;
+      return (
+        renderError?.(error) || (
+          <div className={errorClassName || null}>Render Failure: {error.message}</div>
+        )
+      );
```

Comment:
Optional chaining is nice in many cases but this one seems like adds more complexity than it saves. This code will also render the fallback if `renderError(error)` returns a falsey value, which seems like confusing behavior. 

```suggestion
      return (
        renderError ? renderError(error) : (
          <div className={errorClassName || null}>Render Failure: {error.message}</div>
        )
      );
```